libc/fuzzing/__support/hashtable_fuzz.cpp

```diff
@@ -0,0 +1,159 @@
+#include "src/__support/CPP/new.h"
+#include "src/__support/CPP/optional.h"
+#include "src/__support/HashTable/table.h"
+#include "src/string/memcpy.h"
+#include <search.h>
+#include <stdint.h>
+namespace LIBC_NAMESPACE {
+
+enum class Action { Find, Insert, CrossCheck };
+static uint8_t *global_buffer = nullptr;
```

Comment:
why did you do this using a global buffer? This seems like it'd be much simpler if it was just a buffer allocated in the fuzzing function and deleted when the function ends, possibly using its own class. That way you also wouldn't need to have your own cleanup mechanism.

Reply:
I need to inject zero bytes for string termination.

Reply:
look at the implementation of strtofloat_fuzz. There we allocate a new buffer inside of the function so we can add a null terminator and do other string modifications. It doesn't need to be global, it just needs to live as long as the function call.

Reply:
It is a bit different here as the lifetime of key and values should be the lifetime of the hashtable. I will look into a better way.